canvas_sdk/effects/base.py

```diff
@@ -1,21 +1,18 @@
 import json
 from typing import Any
 
-from pydantic import BaseModel, ConfigDict
-
+from canvas_sdk.base import Model
 from canvas_sdk.effects import Effect, EffectType
 
 
-class _BaseEffect(BaseModel):
+class _BaseEffect(Model):
```

Comment:
Given the change in direction we're taken with the data module, we should have a discussion as a group about how effects should be modeled, and what the user experience should be. Previously there was an assumption that effects would be generated off of data objects that were mutated, but that does not appear to be the case any longer with the Django-based approach.

Questions I might ask:

* Are we getting any real value out of Pydantic when one considers that the effect classes are responsible for creating the payloads, not users?
* Would the user get more value out of tightly-defined interfaces that produce IDE warnings vs. getting validation warnings at runtime?
* What should the user experience of creating and returning an effect look like? Is there any need to make them call methods on effects (like `effect_payload`) as opposed to just having users create and return the effect objects from `compute`?

Reply:
pydantic gives us some nice ide suggestions, validation methods, and error-handling (at runtime). at this point i don't feel like its causing much harm, but if anybody thinks that's the case, or that its too burdensome we can definitely move away. i've been happy with it so far. 

so far, the developer should only ever use the `.apply` method from effects. everything else is internal. maybe i should prepend a `_` to all the other methods...

Reply:
Yeah, it gives you autocomplete in terms of the fields on the model, but it doesn't necessarily tell you what fields have to be provided in order for the effect to be valid. You won't know that until runtime, but with a well-designed interface, you could get warnings in your IDE while you are creating the effect.